### PR TITLE
Sus 180 remove duplicate code in walls

### PR DESCRIPTION
**Pull request by sqreek**
follow up from https://github.com/Wikia/app/pull/9716

Mostly remove duplicate Extract WallNotification and GlobalNavigationWallNotification duplicate code into base controller

// cc @Wikia/sustaining-team

### CODE REVIEW

File: extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+			$this->forward( __CLASS__, 'NotificationAdmin' );
+			return;
+		}
+
+		$authors = [ ];
+		foreach ( $notify[ 'grouped' ] as $notify_entity ) {
+			$authors[] = [
+				'displayname' => $notify_entity->data->msg_author_displayname,
+				'username' => $notify_entity->data->msg_author_username,
+				'avatar' => AvatarService::renderAvatar(
```
Comment (sqreek) [inline extensions/wikia/WallNotifications/WallNotificationControllerBase.class.php:102]: !Aware! This part was in GlobalNavigationWallNotificationsController but not in WallNotificationsController

**Comment by ghost** [comment]
Let's give it a shot :+1:

File: extensions/wikia/Wall/WallBaseController.class.php [hunk 2]
```diff
@@ -405,7 +405,7 @@ public function getThreads( $title, $page, $perPage = null ) {
 
 		$this->wall = $this->getWallForIndexPage( $title );
 
-		/* @var $this->wall Wall */
+		/* @var $this ->wall Wall */
```
Comment (kamilkoterba) [inline extensions/wikia/Wall/WallBaseController.class.php]: wanted `/* @var Wall $this->wall */` here?

File: extensions/wikia/Wall/WallHooksHelper.class.php [hunk 3]
```diff
@@ -540,80 +554,77 @@ static public function onPageHeaderIndexAfterActionButtonPrepared( $response, $n
...
+		$parts = explode( '/', $title->getText() );
+		$action = $response->getVal( 'action' );
+		$dropdown = $response->getVal( 'dropdown' );
+		$canEdit = $app->wg->User->isAllowed( 'editwallarchivedpages' );
 
-				if ( $canEdit ) {
-					$action['text'] = wfMessage( 'edit' )->text();
-					$action['id'] = 'talkArchiveEditButton';
-				}
+		if ($title->isSubpage() && !empty( $parts[ 1 ] )){
```
Comment (kamilkoterba) [inline extensions/wikia/Wall/WallHooksHelper.class.php]: spaces within brackets :)

File: extensions/wikia/ArticleComments/classes/ArticleCommentInit.class.php [hunk 4]
```diff
@@ -41,6 +41,8 @@ static public function ArticleCommentCheck( $title = null ) {
 
 	/**
 	 * Check whether comments should be enabled for given title
+	 * @param Title $title
+	 * @return bool
 	 */
 	static public function ArticleCommentCheckTitle( $title ) {
```
Comment (kamilkoterba) [inline extensions/wikia/ArticleComments/classes/ArticleCommentInit.class.php:47]: Are you planning to add strong typing to parameters as you're adding them to docs?

File: extensions/wikia/Wall/WallMessage.class.php [hunk 5]
```diff
@@ -513,8 +515,7 @@ public function setNotifyEveryone( $notifyEveryone ) {
 			if ( $notifyEveryone ) {
 				$this->getArticleComment()->setMetaData( 'notify_everyone', time() );
 				$this->doSaveMetadata( $app->wg->User,
-					wfMessage( 'wall-message-update-highlight-summary' )->inContentLanguage()->text(),
-					false, true );
+					wfMessage( 'wall-message-update-highlight-summary' )->inContentLanguage()->text(), false );
```
Comment (kamilkoterba) [inline extensions/wikia/Wall/WallMessage.class.php]: false is default val for $force param

File: extensions/wikia/Wall/WallMessage.class.php [hunk 6]
```diff
@@ -523,8 +524,7 @@ public function setNotifyEveryone( $notifyEveryone ) {
 				$pageId = $this->getId();
 				$wne->removeNotificationForPageId( $pageId );
 				$this->doSaveMetadata( $app->wg->User,
-					wfMessage( 'wall-message-update-removed-highlight-summary' )->inContentLanguage()->text(),
-					false, true );
+					wfMessage( 'wall-message-update-removed-highlight-summary' )->inContentLanguage()->text(), false );
```
Comment (kamilkoterba) [inline extensions/wikia/Wall/WallMessage.class.php]: false is default val for $force param

File: extensions/wikia/WallNotifications/GlobalNavigationWallNotificationsController.class.php [hunk 7]
```diff
@@ -1,305 +1,54 @@
...
 	const NOTIFICATION_TITLE_LIMIT = 48;
 
 	private $wallHelper;
 
 	public function __construct() {
+		parent::__construct();
 		$this->app = F::App();
 	}
 
 	public function Index() {
```
Comment (kamilkoterba) [inline extensions/wikia/WallNotifications/GlobalNavigationWallNotificationsController.class.php:19]: is there any valid reason for leaving Index method and Update not?

**Comment by kamilkoterba** [comment]
Added some optional comments and questions that you may want to address, but
LGTM!

And can we rerun tests for this branch? It passes on devbox for me.

File: extensions/wikia/ArticleComments/classes/ArticleCommentInit.class.php (hunk 4 above)
Comment (sqreek) [inline extensions/wikia/ArticleComments/classes/ArticleCommentInit.class.php:47]: yes, in next iteration

File: extensions/wikia/WallNotifications/GlobalNavigationWallNotificationsController.class.php (hunk 7 above)
Comment (sqreek) [inline extensions/wikia/WallNotifications/GlobalNavigationWallNotificationsController.class.php:19]: There is no valid reason at all.
It was my sloppy mistake that show us that we cannot trust in tests.
Thank you for pointing that

**Comment by sqreek** [comment]
Thank you @kamilkoterba for good PR. Because I missed that Update function, I was removed code from BaseController after moving it there. 
We should add more tests for this section, because they should failed on this situation

**Comment by ghost** [comment]
Yeah. :+1 - I missed the `Update` function. [Given enough eyeballs, all bugs are shallow.](https://en.wikipedia.org/wiki/Linus%27s_Law).

**Pull request merged**